### PR TITLE
128 remastering contract cards

### LINKED ISSUE #126: Bug: Clicking the text on a contract card doesn't open the details-view.

### LINKED ISSUE #128: Remastering contract cards
## Description
As a stakeholder I want the contract cards to be on-par with the rest of the design so that my customer have a good experience using the website.

## Acceptance criteria
- [x] Contract cards have a reasonable size
- [x] Contract cards are equally large regardless of their images
- [x] Pressing anywhere (except on the favorite button) on the contract-card opens the details view.
- [x] Text on the contract cards shouldn't surpass the borders.

### PR DESCRIPTION
**Pull request by jonathan-lindqvist**
## Summary
Contract cards have a unifrom size. The amount of cards on a row change depending on the screen size, and if the title is too long then ellipsis are used.

image

## Definition of done
- [x] All acceptance criteria are completed.
- [x] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [x] The code is reviewed and accepted by (at least) two other team members.
- [x] Views adhere to design style.

Closes #128 
Closes #126

### CODE REVIEW

**Review by ellsib — CHANGES_REQUESTED** [review]
Looks good overall! 
One thought: it looks quite boring with both white background and white border (where name is). Could we maybe change from white in one of the places? Eg. have the blue theme on the cards to make it look nicer. I know it's not your priority in this story, but we said to improve the look of all parts we work with :)

File: src/Client/Pages/Contracts/ContractCard.razor [hunk 1]
```diff
@@ -6,13 +6,26 @@
 @inject IJSRuntime _js
 @inject HttpClient _http
 
-<div class="card" style="margin-top: 1em;">
-    <img id="contract-thumbnail" src="@Contract.SupplierLogoImagePath" class="img-thumbnail img-fluid" data-bs-toggle="modal" data-bs-target="#@("id_" + Contract.Id)" @onclick="AddToRecentlyViewed"/>
-    <div class="card-body d-flex bd-highlight mb-3 d-flex align-items-center">
-        <h3 class="p-2 bd-highlight flex-grow-1" id="contract-name">@Contract.Name</h3>
-        <FavoriteButton @ref="_favoriteButton" Contract="@Contract" OnFavoriteChange="@FavoriteChange"></FavoriteButton>
-    </div>
+<div class="card ms-2 me-2" style="margin-top: 1em; width: 25rem; height: 18rem;  background-color: #222939;">
```
Comment (martinjonsson01) [inline src/Client/Pages/Contracts/ContractCard.razor:9]: nit:
It'd be a lot more readable if you removed all the inline-styling on all the elements in this file and placed it in a `ContractCard.razor.css` component-scoped stylesheet.

File: src/Client/Pages/Contracts/FavoriteButton.razor [hunk 2]
```diff
@@ -4,16 +4,16 @@
 @using System.Text
 @inject HttpClient _http
 
-<button class="btn ms-auto p-2 flex-shrink-0" @onclick="ChangeFavorite">
+<button class="btn p-0" style="color:white;" @onclick="ChangeFavorite">
```
Comment (martinjonsson01) [inline src/Client/Pages/Contracts/FavoriteButton.razor:7]: nit:
Same in this component.

**Review by martinjonsson01 — APPROVED** [review]
lgtm 👍

**Comment by martinjonsson01** [comment]
@jonathan-lindqvist I recall you mentioning something related to #126, have you resolved that in this PR? If so, could you add it as one of the issues that get closed when this is merged?

**Pull request merged**